package/gluon-web-admin/luasrc/lib/gluon/config-mode/model/admin/remote.lua

```diff
@@ -76,29 +75,12 @@ function pw2.cfgvalue()
 end
 
 local function set_password(password)
-	local inr, inw = unistd.pipe()
-	local pid = unistd.fork()
-
-	if pid < 0 then
-		return false
-	elseif pid == 0 then
-		unistd.close(inw)
-
-		local null = fcntl.open('/dev/null', fcntl.O_WRONLY)
-		unistd.dup2(null, unistd.STDOUT_FILENO)
-		unistd.dup2(null, unistd.STDERR_FILENO)
-		if null > 2 then
-			unistd.close(null)
-		end
-
-		unistd.dup2(inr, unistd.STDIN_FILENO)
-		unistd.close(inr)
-
-		unistd.execp('passwd', {[0] = 'passwd'})
-		os.exit(127)
-	end
+	local policies = {[unistd.STDIN_FILENO]=util.PipePolicies.CREATE,
+			  [unistd.STDOUT_FILENO]=util.PipePolicies.DISCARD,
+			  [unistd.STDERR_FILENO]=util.PipePolicies.DISCARD}
```

Comment:
`[unistd.STDOUT_FILENO]=util.PipePolicies.DISCARD` etc. is fairly verbose. How about changing to string keys `stdout = ...`?

Together with the changes suggested about, it would even be possible to merge the `command` and `policies`/`fds` arguments and make the function even more similar to Python's `subprocess`:

```lua
local pid, pipes = util.subprocess.popen {'passwd', stdin=util.subprocess.PIPE, stdout=util.subprocess.DEVNULL, stderr=util.subprocess.DEVNULL}
local inw = pipes.stdin
```

Reply:
You mean like this?

```suggestion
	local policies = {["stdin"]=util.subprocess.PIPE,
			  ["stdout"]=util.subprocess.DEVNULL,
			  ["stderr"]=util.subprocess.DEVNULL}
```

Reply:
Well that didn't handle the last commits well. Ill add the proposal below.

Reply:
https://github.com/freifunk-gluon/gluon/pull/2244#discussion_r667534113